src/resolve/resolved-dns-scope.c

```diff
@@ -793,15 +793,8 @@ DnsTransaction *dns_scope_find_transaction(DnsScope *scope, DnsResourceKey *key,
         /* Try to find an ongoing transaction that is a equal to the
          * specified question */
         t = hashmap_get(scope->transactions_by_key, key);
-        if (!t) {
-                DnsResourceKey *key_any;
-
-                key_any = dns_resource_key_new(DNS_CLASS_IN, DNS_TYPE_ANY, dns_resource_key_name(key));
-                t = hashmap_get(scope->transactions_by_key, key_any);
-                key_any = dns_resource_key_unref(key_any);
-                if (!t)
-                        return NULL;
-        }
+        if (!t)
```

Comment:
The mDNS RFC recommends probing RRs by sending requests with their type set to `ANY`. With this revert no mDNS probing transactions will ever succeed and conflict resolution will get broken since responses have specific types.

But yeah, this logic is mDNS specific and should replace the call to `dns_scope_find_transaction()` in `resolved-mdns.c`.